extends Area3D
class_name InteractionArea
## äº¤äº’åŒºåŸŸ - æ£€æµ‹çŒ«å’ªè¿›å…¥ç‰¹å®šåŒºåŸŸ

@export var interaction_name: String = "æœªå‘½ååŒºåŸŸ"
@export var interaction_type: String = "generic"

signal cat_entered(area_name: String, area_type: String)
signal cat_exited(area_name: String, area_type: String)

var cat_inside: bool = false


func _ready():
	body_entered.connect(_on_body_entered)
	body_exited.connect(_on_body_exited)


func _on_body_entered(body: Node3D):
	if body is Cat3D:
		cat_inside = true
		emit_signal("cat_entered", interaction_name, interaction_type)
		print("ğŸ± çŒ«å’ªè¿›å…¥: ", interaction_name)


func _on_body_exited(body: Node3D):
	if body is Cat3D:
		cat_inside = false
		emit_signal("cat_exited", interaction_name, interaction_type)
		print("ğŸ± çŒ«å’ªç¦»å¼€: ", interaction_name)

